### PR TITLE
Apply Raft node management fixes

### PR DESCRIPTION
**Pull request by metanet**
Please see individual commit messages for details.

Fixes https://github.com/hazelcast/hazelcast-enterprise/issues/3256

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/cp/internal/raft/impl/RaftNodeStatus.java [hunk 1]
```diff
@@ -49,15 +49,17 @@
...
+     * is completed, then the statuses become {@link #TERMINATED}.
      */
     TERMINATING,
 
     /**
-     * When a Raft group is terminated completely, all Raft nodes' statuses in
-     * that group become {@code TERMINATED}.
+     * When a Raft group is destroyed, all Raft nodes' statuses in that group
+     * become {@code TERMINATED} after the destroy process is done. Moreover,
+     * a Raft node can move to {@code TERMINATED} status if its JVM is being
```
Comment (mdogan) [inline hazelcast/src/main/java/com/hazelcast/cp/internal/raft/impl/RaftNodeStatus.java]: It doesn't have to be JVM shutting down. This should be, "... if the `HazelcastInstance` which `RaftNode` belongs to shutting down."

Comment (metanet) [reply hazelcast/src/main/java/com/hazelcast/cp/internal/raft/impl/RaftNodeStatus.java]: That is correct. I didn't want to refer to `HazelcastInstance` from the core Raft code Javadoc but we already have leaks in a few places so I can refer to `HazelcastInstance` here as well.

Comment (mdogan) [reply hazelcast/src/main/java/com/hazelcast/cp/internal/raft/impl/RaftNodeStatus.java]: Or you can just say Hazelcast member/node without linking the class.

**Comment by metanet** [comment]
run-lab-run

**Pull request merged**